internal/tsc_wrapped/tsconfig.ts

```diff
@@ -276,6 +277,38 @@ export function parseTsconfig(
           devmodeTargetOverride: isUndefined(existingBazelOpts.devmodeTargetOverride)
             ? newBazelBazelOpts.devmodeTargetOverride
             : existingBazelOpts.devmodeTargetOverride,
+        };
+
+        // merge compilerOptions plugins @bazel/tsetse disabledRules
+        const existingCompilerOptions: ts.CompilerOptions = existingConfig.compilerOptions || {};
```

Comment:
we need a unit test for this file...

Reply:
I added this:
```
  it('honors plugins config in users tsconfig', () => {
    const userTsconfig = {
      compilerOptions: {
        plugins: [{
          name: '@bazel/tsetse',
          disabledRules: ['must-use-promises'],
        }]
      }
    };
    const generatedTsconfig = {
      extends: './user.tsconfig',
      files: ['a.ts'],
      compilerOptions: {},
    };
    const files = {
      [resolveNormalizedPath('path/to/user.tsconfig.json')]:
          JSON.stringify(userTsconfig),
      [resolveNormalizedPath('path/to/generated.tsconfig.json')]:
          JSON.stringify(generatedTsconfig),
    };
    const host = mockHost(files);
    const [parsed, diagnostics, {target}] =
        parseTsconfig('path/to/generated.tsconfig.json', host);
    expect(diagnostics).toBeNull();
    if (!parsed) {
      fail('Expected parsed');
    } else {
      const {options, bazelOpts, files, config} = parsed;
      if (!options) {
        fail('no options');
      } else {
        const plugins = options['plugins'] as PluginImportWithConfig[] || [];
        expect(plugins.length).toBe(1);
        const [tsetsePlugin] = plugins;
        expect(tsetsePlugin.name).toBe('@bazel/tsetse');
        const disabledRules = tsetsePlugin['disabledRules'] as Array<string>;
        expect(disabledRules.length).toBe(1);
        expect(disabledRules[0]).toBe('must-use-promises');
      }
    }
  });
```
but it passes without your change.

Does it need to be 
generatedTsconfig extends userA extends userB
with userB defining the disabledRules?

That doesn't seem to be what happens in https://github.com/angular/angular/pull/31019/files#diff-36fb2cb41cec12240538690af845c47b so I'm confused